### PR TITLE
feat: Cohere chat models integration

### PR DESCRIPTION
**Pull request by tcapelle**
Cohere API integration with Weave

### CODE REVIEW

**Comment by tcapelle** [comment]
@tssweeney I need help with 2 things:
- the output of the cohere SDK is a `pydantic.v1.BaseModel` that we don't parse correctly
- I think the wrapped function in cohere SDK patches after the weave patch, and that's why the name is ignored. For the streaming version the naming is correctly displayed, I suppose the accumulator fixes the naming.

**Comment by tcapelle** [comment]
The only thing that may need changing is how the patching is applied.

As you know @tssweeney the cohere Client modifies the chat function on the init, so we need to patch before creating a client. Two options for this:
- We call `weave.init` before creating the client (this is weird in general and wouldn't be how most of of the users do, but... it is all over our docs: anthropic, mistral, etc...)
- We manually patch before creating a client. Something closer to where the imports happen.

File: .gitignore [hunk 1]
```diff
@@ -27,4 +27,5 @@ weave/tests/zlib.txt
 weave/tests/execute.json
 weave-js/src/generated
 .integration-deps
+cape_dev/
```
Comment (tssweeney) [inline .gitignore:30]: what is this?

File: docs/docs/guides/integrations/cohere.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
+---
```
Comment (tssweeney) [inline docs/docs/guides/integrations/cohere.md:1]: @scottire may want to review docs as well

File: weave/integrations/cohere/cohere_test.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    )
+
+    exp = response.text
+    assert exp.strip() != ""
+    res = client.server.calls_query(tsi.CallsQueryReq(project_id=client._project_id()))
+    assert len(res.calls) == 1
+    call = res.calls[0]
+    assert call.exception is None and call.ended_at is not None
+    output = _get_call_output(call)
+    assert output.text == exp
```
Comment (tssweeney) [inline weave/integrations/cohere/cohere_test.py:49]: please also add assertions for: `started_at < ended_at`, `ended_at != None`, `exception == null`, and `op_name` are all correct. This comment applies to all tests

File: weave/integrations/file::memory:?cache=shared
Comment (tssweeney) [inline weave/integrations/file::memory:?cache=shared:1]: what is this??

**Review by tssweeney — CHANGES_REQUESTED** [review]
Looks really good!! (: Only a few comments to tend to

**Review by tssweeney — APPROVED** [review]

**Pull request merged**